web-console/src/dialogs/supervisor-table-action-dialog.tsx

```diff
@@ -0,0 +1,165 @@
+import axios from 'axios';
+import * as React from 'react';
+import * as CopyToClipboard from 'react-copy-to-clipboard';
+
+import { AppToaster } from '../singletons/toaster';
+import { downloadFile } from '../utils';
+
+import { SideButtonMetaData, TableActionDialog } from './table-action-dialog';
+
+interface SupervisorTableActionDialogProps extends React.Props<any> {
+  onClose: () => void;
+  metaData: SupervisorTableActionDialogMetaData;
+  terminateSupervisor: (id: string) => void;
+  resetSupervisor: (id: string) => void;
+  resumeSupervisor: (id: string) => void;
+  suspendSupervisor: (id: string) => void;
+}
+
+interface SupervisorTableActionDialogState {
+  jsonValue: string;
+  activeTab: 'payload' | 'status' | 'stats' | 'history';
+}
+
+export interface SupervisorTableActionDialogMetaData {
+  id: string | null;
+  supervisorSuspended: boolean;
+}
+
+export class SupervisorTableActionDialog extends React.Component<SupervisorTableActionDialogProps, SupervisorTableActionDialogState> {
+  constructor(props: SupervisorTableActionDialogProps) {
+    super(props);
+    this.state = {
+      jsonValue: '',
+      activeTab: 'payload'
+    };
+    this.getJsonInfo('payload');
+  }
+
+  private getJsonInfo = async (endpoint: string): Promise<void> => {
+    const { id } = this.props.metaData;
+    if (endpoint === 'payload') endpoint = '';
+    try {
+      const resp = await axios.get(`/druid/indexer/v1/supervisor/${id}/${endpoint}`);
+      const data = resp.data;
+      this.setState({
+        jsonValue: JSON.stringify(data, undefined, 2)
+      });
+    } catch (e) {
+      AppToaster.show({
+        message: e.response.data,
+        intent: Intent.WARNING
+      });
+      this.setState({
+        jsonValue: ''
+      });
+    }
+  }
+
+  private onClickSideButton = (tab: 'payload' | 'status' | 'stats' | 'history') => {
+    this.getJsonInfo(tab);
+    this.setState({
+      activeTab: tab
+    });
+  }
+
+  render(): React.ReactNode {
+    const { onClose, terminateSupervisor, resetSupervisor, resumeSupervisor, suspendSupervisor } = this.props;
+    const { activeTab, jsonValue } = this.state;
+    const { id, supervisorSuspended } = this.props.metaData;
+
+    const supervisorTableSideButtonMetadata: SideButtonMetaData[] = [
+      {icon: 'align-left', text: 'Payload', active: activeTab === 'payload',
+        onClick: () => this.onClickSideButton('payload')},
+      {icon: 'dashboard', text: 'Status', active: activeTab === 'status',
+        onClick: () => this.onClickSideButton('status')},
+      {icon: 'chart', text: 'Statistics', active: activeTab === 'stats',
+        onClick: () => this.onClickSideButton('stats')},
+      {icon: 'history', text: 'History', active: activeTab === 'history',
+        onClick: () => this.onClickSideButton('history')}
+    ];
+
+    return <TableActionDialog
+      isOpen
+      sideButtonMetadata={supervisorTableSideButtonMetadata}
```

Comment:
I would add the "close on escape" here to allow this dialog to be closed with an Esc